examples/dnn_self_supervised_learning_ex.cpp

```diff
@@ -288,73 +289,47 @@ try
         serialize("resnet50_self_supervised_cifar_10.net") << layer<5>(net);
     }
 
-    // To check the quality of the learned feature representations, we will train a linear
-    // classififer on top of the frozen backbone.
-    model::infer inet;
-    // Assign the network, without the projector, which is only used for the self-supervised
-    // training.
-    layer<2>(inet) = layer<5>(net);
-    // Freeze the backbone
-    set_all_learning_rate_multipliers(layer<2>(inet), 0);
-    // Train the network
-    {
-        dnn_trainer<model::infer, adam> trainer(inet, adam(1e-6, 0.9, 0.999), gpus);
-        // Since this model doesn't train with pairs, just single images, we can increase
-        // the batch size.
-        trainer.set_mini_batch_size(2 * batch_size);
-        trainer.set_learning_rate(learning_rate);
-        trainer.set_min_learning_rate(min_learning_rate);
-        trainer.set_iterations_without_progress_threshold(5000);
-        trainer.set_synchronization_file("cifar_10_sync");
-        trainer.be_verbose();
-        cout << trainer << endl;
-
-        std::vector<matrix<rgb_pixel>> images;
-        std::vector<unsigned long> labels;
-        while (trainer.get_learning_rate() >= trainer.get_min_learning_rate())
-        {
-            images.clear();
-            labels.clear();
-            while (images.size() < trainer.get_mini_batch_size())
-            {
-                const auto idx = rnd.get_random_32bit_number() % training_images.size();
-                images.push_back(augment(training_images[idx], false, rnd));
-                labels.push_back(training_labels[idx]);
-            }
-            trainer.train_one_step(images, labels);
-        }
-        trainer.get_net();
-        inet.clean();
-        serialize("resnet50_cifar_10.dnn") << inet;
-    }
+    // Now, we initialize the feature extractor model with the backbone we have just learned.
+    model::feats fnet(layer<5>(net));
+    // And we will generate all the features for the training set to train a multiclass SVM
+    // classifier.
+    std::vector<matrix<float, 0, 1>> features;
+    cout << "Extracting features for linear classifier..." << endl;
+    features = fnet(training_images, 4 * batch_size);
+    svm_multiclass_linear_trainer<linear_kernel<matrix<float,0,1>>, unsigned long> trainer;
+    trainer.set_num_threads(std::thread::hardware_concurrency());
+    trainer.set_c(svm_c);
```

Comment:
```suggestion
    // The most appropriate C setting could be found automatically by using find_max_global().  See the docs for
    // find_max_global() for further information and take particular note of model_selection_ex.cpp.
    trainer.set_c(svm_c);
```

Reply:
IDK, you could add find_max_global() down below to auto select the right C.  At some point it makes the example too busy though.  Each example doesn't (and shouldn't) do everything possible.  I think leaving a comment letting someone know they can is the right thing in this case though, but I could maybe go either way.

Reply:
Good idea, I will try to find the optimal C using `find_max_global` and set it to that value by default, and add the comment.